### PR TITLE
cmake-config fixes

### PR DESCRIPTION
**Pull request by bertwesarg**

### CODE REVIEW

File: dash/dash-config.cmake.in [hunk 1]
```diff
@@ -8,14 +8,14 @@
...
-set(DASH_VERSION_MAJOR "@DASH_VERSION_MAJOR@")
-set(DASH_VERSION_MINOR "@DASH_VERSION_MINOR@")
-set(DASH_VERSION_PATCH "@DASH_VERSION_PATCH@")
+set(DASH_VERSION_MAJOR "@DASH_VERSION_MAJOR@")
+set(DASH_VERSION_MINOR "@DASH_VERSION_MINOR@")
+set(DASH_VERSION_PATCH "@DASH_VERSION_PATCH@")
 
 set(DASH_LIBRARY "@DASH_LIBRARY@")
 set(DASH_LIBRARIES ${DASH_LIBRARY} "dart-@dart_variant@")
 set(DASH_CXX_FLAGS "@VARIANT_ADDITIONAL_COMPILE_FLAGS@ @CMAKE_CXX_FLAGS_RELEASE@")
```
Comment (fmoessbauer) [inline dash/dash-config.cmake.in:17]: Currently we always export the release target and there is no possibility to install a debug build of DASH. IMO we should export the current target. I already fixed this [here](https://github.com/dash-project/dash/blob/feat-coarray/dash/dash-config.cmake.in#L17). However this is still not optimal as some flags (e.g. OpenMP) are still not exported.

While you are on this, can you provide a fix?

Comment (bertwesarg) [reply dash/dash-config.cmake.in:17]: No way. Its not actually clear to me, how these cmake files should be used at all. While `find_package` works, I still need ugly ways to extract the flags from the package and add them to my target. I would have expected, that a simple `add_dependency(my-dash-app dash-mpi)` suffice, but its not. See [here](https://github.com/mephisto-hpc/testing/blob/master/CMakeLists.txt#L65) for my current approach.

Comment (fmoessbauer) [reply dash/dash-config.cmake.in:17]: Ok, that looks really complicated. I wrote once a simple [dummy-app](https://github.com/dash-project/dash-apps/blob/master/dummy-app/CMakeLists.txt) showing how to include DASH using CMake. Perhaps that helps.

There is also an article (I do not find it currently..) from the GoogleTest guys on how they recommend to include gtest.

Comment (bertwesarg) [reply dash/dash-config.cmake.in:17]: [This is getting offtopic for this PR]

Ok, I must have been running in circles, just using `target_link_libraries()` with `${DASH_LIBRARIES}` is sufficient. But do I need to use the CXX flags from DASH too?

Comment (fmoessbauer) [reply dash/dash-config.cmake.in:17]: With the changes I made to the config file, I recommend to use it. Many of the flags represent the cmake configuration of the dash build (e.g. use hwloc,...). So if not using them, you might run into problems as the headers might not fit to the installed library. However currently even the optimization flags are included. That is what I fixed in the referenced commit one comment above.

**Review by fmoessbauer — APPROVED** [review]

**Pull request merged**